Controller/Adminhtml/Bread/ValidateCredentials.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Bread\BreadCheckout\Controller\Adminhtml\Bread;
+
+/**
+ * Class ValidateCredentials
+ * @package Bread\BreadCheckout\Controller\Adminhtml
+ */
+class ValidateCredentials extends \Magento\Backend\App\Action
+{
+
+    const API_SANDBOX_URI = 'https://api-sandbox.getbread.com/carts/';
+    const API_LIVE_URI    = 'https://api.getbread.com/carts/';
+
+    /**
+     * @var \Bread\BreadCheckout\Helper\Log
+     */
+    public $logger;
+
+    public function __construct(
+        \Magento\Backend\App\Action\Context $context,
+        \Bread\BreadCheckout\Helper\Log $log
+    )
+    {
+        $this->logger = $log;
+        parent::__construct($context);
+    }
+
+    public function execute()
+    {
+
+        $params = $this->getRequest()->getParams();
+        $result = $this->testCredentials($params['apiMode'],$params['pubKey'],$params['secKey']);
+
+        return $this->resultFactory->create(\Magento\Framework\Controller\ResultFactory::TYPE_JSON)->setData($result);
+    }
+
+    private function testCredentials($apiMode,$username,$password)
+    {
+        $dummy = [];
+        $dummy['expiration'] = date('Y-m-d');
+        $dummy['options'] = [];
+        $dummy['options']['customTotal'] = 10000;
```

Comment:
can you add in the cartname here too? I think you added it for the M1 version but might have missed it here

Reply:
Yes missed it, just pushed